src/convertToObject.js

```diff
@@ -6,7 +6,36 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString.length === 0) {
+    return {};
+  }
+
+  const newStyles = sourceString.trim();
+  const stylesArray = newStyles.split(';');
+  const arr = [];
+
+  for (let i = 0; i < stylesArray.length; i++) {
+    if (stylesArray[i].trim().length !== 0) {
+      arr.push(stylesArray[i].replace('\n', ''));
+    }
+  }
+
+  const resultArray = [];
+
+  for (let i = 0; i < arr.length; i++) {
+    const part = arr[i].split(': ');
+
+    resultArray.push(part);
```

Comment:
[CODE PRACTICE] - Again, avoid using loops. You can chain `map` directly after the `filter` to transform the array items into the desired format.